statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,721 @@
+        """
+        Generic Zero Inflated model Hessian matrix of the loglikelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        hess : ndarray, (k_vars, k_vars)
+            The Hessian, second derivative of loglikelihood function,
+            evaluated at `params`
+
+        Notes
+        -----
+        """
+        hess_arr_main = self._hessian_main(params)
+        hess_arr_infl = self._hessian_inflate(params)
+
+        if hess_arr_main is None or hess_arr_infl is None:
+            return approx_hess(params, self.loglike)
+
+        dim = self.k_exog + self.k_inflate
+
+        hess_arr = np.zeros((dim, dim))
+
+        hess_arr[:self.k_inflate,:] = hess_arr_infl
+        hess_arr[self.k_inflate:,self.k_inflate:] = hess_arr_main
+
+        tri_idx = np.triu_indices(self.k_exog + self.k_inflate, k=1)
+        hess_arr[tri_idx] = hess_arr.T[tri_idx]
+
+        return hess_arr
+
+    def predict(self, params, exog=None, exog_infl=None, exposure=None,
+                offset=None, which='mean'):
+        """
+        Predict response variable of a count model given exogenous variables.
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+        exog : array, optional
+            A reference to the exogenous design.
+            If not assigned, will be used exog from fitting.
+        exog_infl : array, optional
+            A reference to the zero-inflated exogenous design.
+            If not assigned, will be used exog from fitting.
+        offset : array, optional
+            Offset is added to the linear prediction with coefficient equal to 1.
+        exposure : array, optional
+            Log(exposure) is added to the linear prediction with coefficient
+            equal to 1. If exposure is specified, then it will be logged by the method.
+            The user does not need to log it first.
+        which : string, optional
+            Define values that will be predicted.
+            'mean', 'poisson-mean', 'linear', 'mean-nonzero', 'prob-zero, 'prob'
+            Default is 'mean'.
+
+        Notes
+        -----
+        """
+        if exog is None:
+            exog = self.exog
+            offset = getattr(self, 'offset', 0)
+            exposure = getattr(self, 'exposure', 0)
+
+        if exog_infl is None:
+            exog_infl = self.exog_infl
+
+        if exposure is None:
+            exposure = 0
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = 0
+
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        if self.infl == 'logit':
+            prob_poisson = 1 / (1 + np.exp(np.dot(exog_infl, params_infl)))
+        elif self.infl == 'probit':
+            raise NotImplemented('Predict for Probit inflation not implemented')
```

Comment:
In both cases,
`self.model_infl.predict(params, exog_infl)`
should provide the correct results, for the predicted probability

Reply:
All tests for predict failed.

Reply:
I think `prob_main = 1 - self.model_infl.predict(params_infl, exog_infl)`
the infl predict is the prob of inflation, AFAICS now